src/App/Pages/Accounts/LoginPasswordlessPage.xaml.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace Bit.App.Pages
+{
+    public partial class LoginPasswordlessPage : BaseContentPage
+    {
+        private LoginPasswordlessViewModel _vm;
+
+        public LoginPasswordlessPage(string fingerprintPhrase, string email, string deviceType, string ipAddress, string location, string origin, DateTime requestDate)
+        {
+            InitializeComponent();
+            _vm = BindingContext as LoginPasswordlessViewModel;
+            _vm.Page = this;
+
+            _vm.LoginRequest = new LoginPasswordlessDetails()
+            {
+                Origin = origin,
+                Email = email,
+                DeviceType = deviceType,
+                IpAddress = ipAddress,
+                NearLocation = location,
+                FingerprintPhrase = fingerprintPhrase,
+                RequestDate = requestDate,
+            };
```

Comment:
What I meant on the previous request was to change the constructor so it's more direct and easier to maintain if one parameter is added/modified/removed, i.e.:
```c#
public LoginPasswordlessPage(LoginPasswordlessDetails loginRequest)
{
    InitializeComponent();
    _vm = BindingContext as LoginPasswordlessViewModel;
    _vm.Page = this;
    _vm.LoginRequest = loginRequest;
```
In this manner, if anything changes, you only have to change the `LoginPasswordlessDetails` and update the bindings and that's it. In the current way you'd have to change the constructor, the assignment here and the bindings and any place that is calling the constructor

Reply:
oh ye it makes total sense.